### ISSUE TITLE
dotnet --version completes with exit code 0 when global.json is malformed

### ISSUE THREAD

**Issue by matkoch** [dotnet/runtime#94496]
### Describe the bug
`dotnet --version` completes fine even when `global.json` is malformed JSON.

### To Reproduce
Make `global.json` malformed and execute `dotnet --version`.

### Exceptions (if any)
Should exit with non-zero exit code. Instead:

```
$ dotnet --version
A JSON parsing exception occurred in [/Users/matt/code/nuke/external/functions/global.json], offset 36 (line 3, column 24): Invalid encoding in string.
6.0.100

$ echo $?
0
```

### Further technical details
MacOS

**Comment by vrushalivartak** [comment]
dotnet --info command is dependent on the global_json_path returned by hostfxr_resolve_sdk2 callback function. 
https://github.com/dotnet/runtime/blob/71a7d8844cf509717f7e76baafb1bfbf1217ceec/src/native/corehost/fxr/command_line.cpp#L327 

And global_json_path value holds the path to global.json only if the global.json is used. Thus in the scenario, when global.json is malformed, it's path is not available in global_json_path.
https://github.com/dotnet/runtime/blob/71a7d8844cf509717f7e76baafb1bfbf1217ceec/src/native/corehost/fxr/hostfxr.cpp#L217 

Hence 4th key would need to be defined to be returned in hostfxr_resolve_sdk2_result_fn. The value of which will hold the path to global.json which was found but malformed or invalid.
Key name something like,
invalid_global_json_path

This should have no impact any other existing functionality. And invalid_global_json_path can be used to print the invalid file path in the output of dotnet --info.

@elinor-fung Would you agree?

**Comment by vitek-karas** [comment]
Don't we just need to update this:
https://github.com/dotnet/runtime/blob/af0227fad06abc47fe109960a695054b0dcdd473/src/native/corehost/fxr/command_line.cpp#L324-L327

**Comment by vrushalivartak** [comment]
command_line.cpp update alone is not enough because the global_json_path will be empty.

global_json_path value holds the path to global.json only if the global.json is used. Thus in the scenario, when global.json is malformed, it's path is not available in global_json_path.
Refer hostfxr_resolve_sdk2(..) function comment,
https://github.com/dotnet/runtime/blob/af9e24526fcd3392227a734509598cf59f6eceb6/src/native/corehost/fxr/hostfxr.cpp#L217-L223

The global_json_path value is set if global_file_path() is not empty,
https://github.com/dotnet/runtime/blob/af0227fad06abc47fe109960a695054b0dcdd473/src/native/corehost/fxr/hostfxr.cpp#L275-L280

global_file_path() returns global_file,
https://github.com/dotnet/runtime/blob/af0227fad06abc47fe109960a695054b0dcdd473/src/native/corehost/fxr/sdk_resolver.cpp#L43-L47

global_file is set at the end of parse_global_file(..) function,
https://github.com/dotnet/runtime/blob/af0227fad06abc47fe109960a695054b0dcdd473/src/native/corehost/fxr/sdk_resolver.cpp#L356

In case of malformed global.json, parse_global_file(..) function returns much earlier without setting global_file,
https://github.com/dotnet/runtime/blob/af0227fad06abc47fe109960a695054b0dcdd473/src/native/corehost/fxr/sdk_resolver.cpp#L252-L255

**Comment by vitek-karas** [comment]
Yes - you're right. I mean my comment mostly as "do we really need to expose this information through `hostfxr_resolve_sdk`?" (which I think we don't, at least not for this specific request)
